### PR TITLE
[BackEnd] Added remote books to books list, books initial add, books detail

### LINKED ISSUE #769: Add Remote Inventory Count and Remote Retail Price to Books List

### LINKED ISSUE #770: Add Intrinsic Fields from Remote Subsidiary Store for a specific book on Book Detail
2.2.3. The system should query the subsidiary store about this book by ISBN and show any intrinsic fields that differ on their system versus ours (except cover art). **Remote inventory count** and **remote retail price** should also be shown.

### LINKED ISSUE #771: Add Users to import remote editable intrinsic fields on book initial add
2.3.3.2. The system should query the subsidiary store for any editable intrinsic fields (page count, dimensions, cover art) for the given book(s) and allow the user to easily import those values individually if they so desire.

### LINKED ISSUE #772: Add Users to be able to import intrinsic fields of a book based on remote subsidiary store data
2.4.2. The system should query the subsidiary store for any editable intrinsic fields (page count, dimensions, cover art) for the given book(s) and allow the user to easily import those values individually if they so desire.

### PR DESCRIPTION
**Pull request by hkder**
## Feature Description/Implementation
Closes #769
Closes #770 
Closes #771 
Closes #772

### CODE REVIEW

File: backend/books/views.py [hunk 1]
```diff
@@ -76,18 +78,30 @@ def post(self, request):
...
+        book_data = {}
 
         if isbn in isbn_query_set:
-            shared_dict['books'].append(self.parseDBBookModel(query_set[isbn_query_set.index(isbn)]))
+            book_data = self.parseDBBookModel(query_set[isbn_query_set.index(isbn)])
         else:
-            self.populate_shared_dict_with_isbn_data(isbn, shared_dict)
+            book_data = self.populate_shared_dict_with_isbn_data(isbn, shared_dict)
+        
+        if remote_book_data:
```
Comment (caseyszilagyi) [inline backend/books/views.py:89]: What would happen here if remote_book exists for a given ISBN, but for some reason we can't get it from the external API (Google books)? Seems unlikely, but was just wondering

**Review by caseyszilagyi — CHANGES_REQUESTED** [review]

**Comment by danielfeinblatt** [comment]
After Casey's questions have been addressed, LGTM

File: backend/books/remote_books.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    def get_remote_book_data(self, isbn):
+        url = self._remote_store_api_url
+        headers = self.create_headers()
+        payload = self.create_payload(isbn)
+        response = requests.request("POST", url, headers=headers, data=payload)
+
+        # The first index will be removed if the reponse is only a dictionary
+        ret = response.json()[0]
+        ret = ret[isbn]
```
[the remark this answers is not in the document]
Comment (hkder) [reply backend/books/remote_books.py:19]: this is related to parsing the response group 3 sends us. they send us an array not a dictionary.

**Review by caseyszilagyi — APPROVED** [review]
LGTM

**Pull request merged**